File: Fuera de tema/Grados.py
# Grados.py
# Convierte grados Fahrenheit a Celsius según lo introducido por el usuario.

def f_to_c(fahrenheit: float) -> float:
    return (fahrenheit - 32.0) * 5.0 / 9.0

def main():
    raw = input("Introduce grados Fahrenheit (ej. 73.4): ").strip()
    raw = raw.replace(",", ".")
    try:
        f = float(raw)
    except ValueError:
        print("Entrada no válida. Introduce un número.")
        return
    c = f_to_c(f)
    print(f"{f:.2f} °F son {c:.2f} °C")

if __name__ == "__main__":
    main()

File: Fuera de tema/Grados.java
// Grados.java
// Convierte grados Fahrenheit a Celsius según lo introducido por el usuario.

import java.util.Scanner;

public class Grados {
    public static double fToC(double fahrenheit) {
        return (fahrenheit - 32.0) * 5.0 / 9.0;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Introduce grados Fahrenheit (ej. 73.4): ");
        String raw;
        try {
            raw = sc.nextLine().trim();
        } catch (Exception e) {
            System.out.println("No se pudo leer la entrada.");
            sc.close();
            return;
        }
        raw = raw.replace(',', '.');
        try {
            double f = Double.parseDouble(raw);
            double c = fToC(f);
            System.out.printf("%.2f °F son %.2f °C%n", f, c);
        } catch (NumberFormatException e) {
            System.out.println("Entrada no válida. Introduce un número.");
        } finally {
            sc.close();
        }
    }
}
